File: 4-Increasing-Array/sol.cpp
#include <bits/stdc++.h>
using namespace std;

#define deb(x) cout << #x << "=" << x << endl
#define ll long long
void solve()
{
	ll size;
	cin >> size;
	ll sol = 0,curr,prev = -1;
	cin >> prev;
	for(int i=1;i<size;i++){
		cin >> curr;
		
		if(prev > curr){
			sol += prev - curr;
		}else{
			prev = curr;
		}
		
	}
	cout << sol;
}


int main()
{
	ios_base::sync_with_stdio(0), cin.tie(0), cout.tie(0);
	int t = 1;
	// cin >> t;

	while (t--)
	{
		solve();
	}
	return 0;
}

File: 4-Increasing-Array/sol.java
import java.util.*;
import java.lang.*;
import java.io.*;

class sol{
    // Comparator<> compare = new Comparator<>() {
    //     @Override
    //     public int compare( a, b) {
    //         return;
    //     }
    // };
    public static void  solve(){
        long size,prev,sol = 0,curr;
        Scanner sc=new Scanner(System.in);
        size = sc.nextLong();
        prev = sc.nextLong();
        for(long i = 1 ; i < size; i++){
            curr = sc.nextLong();
            if( prev > curr){
                sol += prev - curr;
            }else{
                prev = curr;
            }
        }

        System.out.println(sol);
    }
    public static void main (String[] args) throws java.lang.Exception
	{
        Scanner sc=new Scanner(System.in);
		int t=1;
        // t = sc.nextInt();
        while(t>0){
            solve();
            t--;
        }
    }
}